vsphere/resource_vsphere_vapp_container.go

```diff
@@ -0,0 +1,369 @@
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(vAppContainerMemorySharesLevelAllowedValues, false),
+			Default:      "normal",
+		},
+		"memory_shares": {
+			Type:        schema.TypeInt,
+			Description: "The number of shares allocated. Used to determine resource allocation in case of resource contention. If this is set, memory_share_level must be custom.",
+			Computed:    true,
+			Optional:    true,
+		},
+		"memory_reservation": {
+			Type:        schema.TypeInt,
+			Description: "Amount of memory (MB) that is guaranteed available to the vApp container.",
+			Optional:    true,
+			Default:     0,
+		},
+		"memory_expandable": {
+			Type:        schema.TypeBool,
+			Description: "Determines if the reservation on a vApp container can grow beyond the specified value, if the parent resource pool has unreserved resources.",
+			Optional:    true,
+			Default:     true,
+		},
+		"memory_limit": {
+			Type:        schema.TypeInt,
+			Description: "The utilization of a vApp container will not exceed this limit, even if there are available resources. Set to -1 for unlimited.",
+			Optional:    true,
+			Default:     -1,
+		},
+		vSphereTagAttributeKey:    tagsSchema(),
+		customattribute.ConfigKey: customattribute.ConfigSchema(),
+	}
+	return &schema.Resource{
+		Create: resourceVSphereVAppContainerCreate,
+		Read:   resourceVSphereVAppContainerRead,
+		Update: resourceVSphereVAppContainerUpdate,
+		Delete: resourceVSphereVAppContainerDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVSphereVAppContainerImport,
+		},
+		Schema: s,
+	}
+}
+
+func resourceVSphereVAppContainerImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client, err := resourceVSphereVAppContainerClient(meta)
+	if err != nil {
+		return nil, err
+	}
+	vc, err := vappcontainer.FromPath(client, d.Id(), nil)
+	if err != nil {
+		return nil, err
+	}
+	d.SetId(vc.Reference().Value)
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceVSphereVAppContainerCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning create", resourceVSphereVAppContainerIDString(d))
+	client, err := resourceVSphereVAppContainerClient(meta)
+	if err != nil {
+		return err
+	}
+	prp, err := resourcepool.FromID(client, d.Get("parent_resource_pool_id").(string))
+	if err != nil {
+		return err
+	}
+	rpSpec := expandVAppContainerConfigSpec(d)
+	vcSpec := &types.VAppConfigSpec{}
+	var f *object.Folder
+	if pf, ok := d.GetOk("parent_folder_id"); ok {
+		f, err = folder.FromID(client, pf.(string))
+		if err != nil {
+			return err
+		}
+	} else {
+		p := strings.Split(prp.InventoryPath, "/")
+		if len(p) > 2 {
+			return fmt.Errorf("unable to locate datacenter name from parent resource pool")
+		}
+		dc, err := getDatacenter(client, p[1])
```

Comment:
This gives me a warning on `vetshadow` as a redundant `err` declaration and can probably be fixed to just be an assignment without declaration (`=` instead of `:=`).

Reply:
I've been ignoring those in cases like this because dc hasn't been declared yet. I considered declaring it on its own line and then just assigning both, but it seemed less clear that way. I'm open to either way.

Reply:
Yeah, declare it. It's a minor inconvenience to get this (and the only other linter hit in this file) gone so that there's eventually less noise in the linter. Might as well have the file completely clean.